openedx/stanford/djangoapps/auth_lagunita/tests/test_api.py

```diff
@@ -0,0 +1,252 @@
+# pylint: disable=missing-docstring
+# pylint: disable=protected-access
+from json import dumps as json_dumps
+import logging
+
+from ddt import ddt, data, unpack
+from django.test import TestCase
+from mock import patch
+from mock import Mock
+import requests
+
+from student.tests.factories import UserFactory
+
+from .utils import MockedResponse
+from ..api import MarketoApi
+
+
+log = logging.getLogger('auth_lagunita')
+
+
+HTTP_CODE_SUCCESS = 200
+VALID_LEAD_ID = 8675309
+VALID_LIST_ID = 666
+DATA = {
+    'access_token': {
+        'access_token': 'foobar',
+    },
+    'error': {
+        'success': False,
+        'errors': [
+            {
+                'code': 1013,
+                'message': 'Object not found',
+            },
+        ],
+        'result': [
+        ],
+    },
+    'exists': {
+        'success': True,
+        'errors': [
+        ],
+        'result': [
+            {
+                'status': 'skipped',
+            },
+        ],
+    },
+    'expired': {
+        'success': False,
+        'errors': [
+            {
+                'code': 602,
+                'message': 'Access token expired',
+            },
+        ],
+        'result': [
+        ],
+    },
+    'missing': {
+    },
+    'nonexisting': {
+        'success': True,
+        'errors': [
+        ],
+        'result': [
+        ],
+    },
+    'success': {
+        'success': True,
+        'errors': [
+        ],
+        'result': [
+            {
+                'status': 'success',
+                'id': VALID_LEAD_ID,
+            },
+        ],
+    },
+    'updated': {
+        'result': [
+            {
+                'id': VALID_LEAD_ID,
+                'status': 'updated',
+            },
+        ],
+        'success': True,
+    },
+}
+RESPONSES = {
+    key: MockedResponse(HTTP_CODE_SUCCESS, json_dumps(value))
+    for key, value in DATA.items()
+}
+RESPONSES.update({
+    'malformed': MockedResponse(HTTP_CODE_SUCCESS, '{"foobar"'),
+    'empty': MockedResponse(HTTP_CODE_SUCCESS, ''),
+    'http_error': MockedResponse(404, ''),
+})
+
+
+class BaseTest(TestCase):
+
+    def setUp(self):
+        super(BaseTest, self).setUp()
```

Comment:
I think this call is superfluous?

Reply:
pylint complains without it

Reply:
Hm, Django doc examples don't make this call, maybe it's a false positive? Not a big deal tho.

Reply:
Yeah, it could also be configuration difference? 🤷‍♂